### PR TITLE
CB-10319 android: Adding reflective helper methods for permission requests

### PR DESCRIPTION
**Pull request by riknoll**
As discussed on the mailing list

### CODE REVIEW

**Comment by riknoll** [comment]
@infil00p @jasongin please review

File: src/android/CameraLauncher.java [hunk 1]
```diff
@@ -26,11 +26,14 @@ Licensed to the Apache Software Foundation (ASF) under one
 import java.io.IOException;
 import java.io.InputStream;
 import java.io.OutputStream;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
```
Comment (jasongin) [inline src/android/CameraLauncher.java]: I think these imports are not needed here. Are they leftover from an earlier iteration?

**Comment by jasongin** [comment]
LGTM, aside from the one minor issue I commented about.

**Comment by omefire** [comment]
Just for context, Here's the link to the related mailing list discussion: http://apache.markmail.org/message/bwhpuh7m6fyf5eem?q=Richard+Knoll+%5BDISCUSS%5D

File: src/android/CameraLauncher.java (hunk 1 above)
Comment (riknoll) [inline src/android/CameraLauncher.java]: They are, I'll update the PR in a bit

**Comment by riknoll** [comment]
I updated/rebased the commit to remove the imports @jasongin mentioned

**Comment by riknoll** [comment]
Tested with manual mobilespec tests on cordova-android's master and cordova-android 4.0.0 on a marshmallow device. The failures in those tests matched the failures in master (Google Photos doesn't work, no camera direction, etc.)

**Comment by riknoll** [comment]
Also tested mobilespec manual tests on a lollipop emulator with cordova-android's master and cordova-android 4.0.0. Again, only expected failures. I'm going to merge this in.

**Pull request merged**